### ISSUE TITLE
Scratchpad position is moving up over original position

### ISSUE THREAD

**Issue by americogobbo** [mypaint/mypaint#597]
My environment is Ubuntu Gnome 14.04.3 with Gnome 3.10
My MyPaint is the Git Master 1.3.0-alpha+git.96101ba

I have this scratchpad to use as drawing pens.
![scratchpad-reverted](https://cloud.githubusercontent.com/assets/8180638/12977026/52882b74-d0af-11e5-818e-37a88d73c735.png)
When I open MyPaint the window is the measure correct, but the scratchpad doc has moved to up.
![scratchpad-after-reopen](https://cloud.githubusercontent.com/assets/8180638/12977057/85361004-d0af-11e5-9950-28e1d85885ec.png)

On terminal we have:
`$:~/installs/mypaint$ ./mypaint`
...
```
(mypaint:2578): Gdk-WARNING **: gdk_window_set_icon_list: icons too large
WARNING: gui.keyboard: Ignoring keybinding for '<Actions>/BrushModifierActions/BlendModeMenu'
INFO: lib.document: Using u'load_ora' to load u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora' (kwargs={'feedback_cb': <function gtk_main_tick at 0x7f9bd4c84b18>})
INFO: lib.document: load_ora: u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
INFO: lib.document: 0.030s load_ora total
INFO: gui.filehandling: Loaded scratchpad from u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
WARNING: gui.accelmap: Not listing '<Actions>/DocumentActions/RotateLeft' (no tooltip, fix before release!)
WARNING: gui.accelmap: Not listing '<Actions>/DocumentActions/RotateRight' (no tooltip, fix before release!)
```

```
(mypaint:2578): Gtk-CRITICAL **: _gtk_tree_view_column_autosize: assertion 'GTK_IS_TREE_VIEW (tree_view)' failed
WARNING: gui.freehand: Add motion event compression workaround: using evhack filter (<TiledDrawWidget object at 0x7f9bd4c0fd20 (TiledDrawWidget at 0x2f39670)>, <gui.freehand.FreehandMode object at 0x7f9bc57b5990>)
WARNING: gui.freehand: Add motion event compression workaround: using evhack filter (<TiledDrawWidget object at 0x7f9bc6f516e0 (TiledDrawWidget at 0x2f398f0)>, <gui.freehand.FreehandMode object at 0x7f9bc5743c50>)
INFO: lib.document: Using u'load_ora' to load u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora' (kwargs={'feedback_cb': <function gtk_main_tick at 0x7f9bd4c84b18>})
INFO: lib.document: load_ora: u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
INFO: lib.document: 0.017s load_ora total
INFO: gui.filehandling: Loaded scratchpad from u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
INFO: gui.drawwindow: Reverted scratchpad to /home/jag/.local/share/mypaint/scratchpads/autosave.ora
```

**Comment by odysseywestra** [comment]
Is this after you close and open up MyPaint again. If that's the case it
could be rounding the scratchpad's canvas size like it dose the main canvas
and center that size instead of the specified size you provide it.

On Thu, Feb 11, 2016, 5:15 AM Americo Gobbo <notifications@github.com>
wrote:

> My environment is Ubuntu Gnome 14.04.3 with Gnome 3.10
> My MyPaint is the Git Master 1.3.0-alpha+git.96101ba
>
> I have this scratchpad to use as drawing pens.
> [image: scratchpad-reverted]
> <https://cloud.githubusercontent.com/assets/8180638/12977026/52882b74-d0af-11e5-818e-37a88d73c735.png>
> When I open MyPaint the window is the measure correct, but the scratchpad
> doc has moved to up.
> [image: scratchpad-after-reopen]
> <https://cloud.githubusercontent.com/assets/8180638/12977057/85361004-d0af-11e5-9950-28e1d85885ec.png>
>
> On terminal we have:
> $:~/installs/mypaint$ ./mypaint
> ...
>
> (mypaint:2578): Gdk-WARNING **: gdk_window_set_icon_list: icons too large
> WARNING: gui.keyboard: Ignoring keybinding for '<Actions>/BrushModifierActions/BlendModeMenu'
> INFO: lib.document: Using u'load_ora' to load u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora' (kwargs={'feedback_cb': <function gtk_main_tick at 0x7f9bd4c84b18>})
> INFO: lib.document: load_ora: u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
> INFO: lib.document: 0.030s load_ora total
> INFO: gui.filehandling: Loaded scratchpad from u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
> WARNING: gui.accelmap: Not listing '<Actions>/DocumentActions/RotateLeft' (no tooltip, fix before release!)
> WARNING: gui.accelmap: Not listing '<Actions>/DocumentActions/RotateRight' (no tooltip, fix before release!)
>
> (mypaint:2578): Gtk-CRITICAL **: _gtk_tree_view_column_autosize: assertion 'GTK_IS_TREE_VIEW (tree_view)' failed
> WARNING: gui.freehand: Add motion event compression workaround: using evhack filter (<TiledDrawWidget object at 0x7f9bd4c0fd20 (TiledDrawWidget at 0x2f39670)>, <gui.freehand.FreehandMode object at 0x7f9bc57b5990>)
> WARNING: gui.freehand: Add motion event compression workaround: using evhack filter (<TiledDrawWidget object at 0x7f9bc6f516e0 (TiledDrawWidget at 0x2f398f0)>, <gui.freehand.FreehandMode object at 0x7f9bc5743c50>)
> INFO: lib.document: Using u'load_ora' to load u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora' (kwargs={'feedback_cb': <function gtk_main_tick at 0x7f9bd4c84b18>})
> INFO: lib.document: load_ora: u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
> INFO: lib.document: 0.017s load_ora total
> INFO: gui.filehandling: Loaded scratchpad from u'/home/jag/.local/share/mypaint/scratchpads/autosave.ora'
> INFO: gui.drawwindow: Reverted scratchpad to /home/jag/.local/share/mypaint/scratchpads/autosave.ora
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mypaint/mypaint/issues/597>.
>
-- 

-- Albert Westra

**Comment by achadwick** [comment]
Doesn't seem to be a problem with the Windows port. I was confused there for a second :smile:

**Comment by americogobbo** [comment]
@achadwick 
Doesn't seem to be a problem with the Windows port. I was confused there for a second :smile: 
Sorry, the plural does much difference :-)

@odysseywestra 
> Is this after you close and open up MyPaint again? If that's the case it could be rounding the scratchpad's canvas size like it dose the main canvas and center that size instead of the specified size you provide it.

Yes.
I have made a test and I've enlarged the window of scratchpad and the problem it seems don't have relation with the window area... with a great area the doc is moving up in the window tab.
![other-area-window-scratchpad](https://cloud.githubusercontent.com/assets/8180638/12977960/2913baa0-d0b5-11e5-8b66-282d69074b68.png

**Comment by odysseywestra** [comment]
One way we could fix this is to add a setting that will give it a frame area to work. Kinda the same way we use a frame on the Canvas itself, but without all the extra GUI. That way the scratchpad can respect the dimensions and Center based on what we give it instead of the rounded tiles size.

@americogobbo just to clarify, dose #647 seem related to your issue?

**Comment by danpla** [comment]
I think I know a possible cause of the problem.

There are lots of Glib.idle_add() in gui/workspace used to set GUI properties, and positions of paneds in particular. When the FileHandler calls scratchpad_doc.reset_view() from open_scratchpad(), initialization of the separators of paneds may not be finished. In such a case, their positions are still set to values calculated automatically on the first "size-allocate" event - they can be smaller or larger than used in a previous session.

In other words, to fix the issue, open_scratchpad() must be called only after GUI is initialed, but I currently don't know how to do this with all of these Glib.idle_add().

**Comment by achadwick** [comment]
Oops, this got closed with #651 as a result of this misfeature in github: https://github.com/blog/1506-closing-issues-via-pull-requests. Sorry, reopened.

**Comment by danpla** [comment]
To clarify. Previously, I wrote:

> ... they can be smaller or larger than used in a previous session.

Actually, the positions are not random.

Previously - before #650 - position if the separator was set to the minimum height of the top child. As a result, the scratchpad image was shifted up.

Now - after #650 - you should see the opposite behavior: the image shifts down, because the separator is set to the height of the parent Gtk.Paned.
